src/main/java/org/jitsi/jicofo/bridge/RegionBasedBridgeSelectionStrategy.java

```diff
@@ -67,11 +91,17 @@ public Bridge doSelect(
         }
 
         return notLoadedAlreadyInConferenceInRegion(bridges, conferenceBridges, participantRegion).orElseGet(
+                () -> notLoadedAlreadyInConferenceInRegionGroup(
+                        bridges, conferenceBridges, getRegionGroup(participantRegion)).orElseGet(
                 () -> notLoadedInRegion(bridges, conferenceBridges, participantRegion).orElseGet(
+                () -> notLoadedInRegionGroup(bridges, conferenceBridges, getRegionGroup(participantRegion)).orElseGet(
                 () -> leastLoadedAlreadyInConferenceInRegion(bridges, conferenceBridges, participantRegion).orElseGet(
+                () -> leastLoadedAlreadyInConferenceInRegionGroup(
+                        bridges, conferenceBridges, getRegionGroup(participantRegion)).orElseGet(
                 () -> leastLoadedInRegion(bridges, conferenceBridges, participantRegion).orElseGet(
+                () -> leastLoadedInRegionGroup(bridges, conferenceBridges, getRegionGroup(participantRegion)).orElseGet(
                 () -> nonLoadedAlreadyInConference(bridges, conferenceBridges, participantRegion).orElseGet(
-                () -> leastLoaded(bridges, conferenceBridges, participantRegion).orElse(null))))));
+                () -> leastLoaded(bridges, conferenceBridges, participantRegion).orElse(null))))))))));
```

Comment:
I'm worried this code is beginning to traverse the same list of bridges over and over again with different criteria, it might be worth thinking about how we could make this more efficient (especially as we start having larger bridge pools).

Reply:
Yeah, if your performance tests don't show anything ridiculous, that's fine.

Reply:
I added some perf tests. On my machine selection from 500 bridges takes about 0.5 ms. I forced the worst-case for the iteration over the lists, though the sorting might be slower in practice. I think that's reasonable.